### DISCUSSION TITLE
How to Change Mandatory fields Required before Successful Social IdP Login?

### DISCUSSION THREAD

**Discussion by abuelgasimsaadeldin** [keycloak/keycloak#8763] [Q&A]
Upon successful login from and Identity Provider (such as Google or Github), the update profile on first login appears with some mandatory fields that need to be filled in namely: **Username**, **Email**, **First Name** and **Last Name**. 

I have added a few extra fields by modifying the "login-update-profile.ftl" file from Keycloak and now would like to add those as Mandatory fields and remove Last Name from being Mandatory? Your help would be highly appreciated. 🙏🙏

Also by default username is pre-filled with the persons username obtained from the IdP, however, at Keycloak I use Phone Number as the unique username, how do I change that field to only pre-fill users phone number (if available) or leave it blank?

Thank you and Best Regards,
Abu

**Comment by jeandenis-k** [comment]
Hi! You can create a login theme, copy over the 'login-update-profile' template from the base Keycloak theme and then remove name fields from the template

**Reply by jeandenis-k** [reply]
Wait, nevermind, that breaks the profile update: it fails if first name and last name are missing 😕
